docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -242,9 +242,224 @@ This guide will help you create a new Expectation Suite by profiling your data w
+
+        To accomplish that aim, it is highly configurable. To describe it's functionality and configuration, we will walk through some of the core concepts followed by examples of increasing complexity. For now, we will use table-based data in this document for simplicity of explanation.
+
+        A Profiler is defined by a set of ProfilerRules that create Expectations for configurable Domains. These ProfilerRules and Expectations can be built using Parameters calculated at runtime, or hard-coded. These Parameters can be local to a domain or global to all domains (QUESTION: should the ``variables`` key be renamed to something more descriptive like ``global_profiler_parameters``?).
+
+        Here is a pseudocode hirearchy to show some of the various core concepts and their relationships (more complex concepts are covered later). Everything prefaced with a ``my_`` is user configurable.
+
+        (QUESTION: Can there be a hirearchy of domains or at least multiple domains for a given rule or should the rule-domain pair be one-one? Maybe domains / parameters can be defined at the same level as rules and then referenced (rather than having to be defined within a rule). Currently it looks like the idea is that more complex domain configurations are meant to be captured in specific domain builder classes e.g. ``SimpleSemanticTypeColumnDomainBuilder`` and that there is a single domain per rule.)
+
+
+        .. code-block:: yaml
+
+          global_domain_builders:
+            - global_domain_builder_1:
+            - global_domain_builder_2:
+            ...
+          global_parameter_builders:
+            - global_param_builder_1:
+            - global_param_builder_2:
+            ...
+          rules:
+            - rule_1
+              rule_specific_domain_builders:
+                - rule_1_domain_builder_1:
+                - rule_1_domain_builder_2:
+                ...
+              rule_specific_parameter_builders:
+                - rule_1_parameter_builder_1:
+                # e.g. reference columns from rule_1_domain_builder_1 to calculate metric on each of them (e.g. ``column.min`` metric)
+                - rule_2_parameter_builder_2:
+                ...
+              expectation_configuration_builders:
+                - expectation_1_builder:
+                # e.g. expect_column_values_to_be_between where domain is set via columns from one of the domains defined above and min/max set via parameters defined above. Can also contain if/then logic.
+
+
+        Here is a slightly more concrete example:
+
+        .. code-block:: yaml
+
+          global_profiler_parameters:
+            - global_param_1
+          rules:
+            my_rule_for_ids: # semantic type
```

Comment:
How do we know that this will be a semantic type?..

Reply:
Good question - we don't, I removed this